lib/Serialization/SerializeDoc.cpp

```diff
@@ -309,53 +310,15 @@ static void writeGroupNames(const comment_block::GroupNamesLayout &GroupNames,
   GroupNames.emit(Scratch, BlobStream.str());
 }
 
-static bool hasDoubleUnderscore(Decl *D) {
-  // Exclude decls with double-underscored names, either in arguments or
-  // base names.
-  static StringRef Prefix = "__";
-
-  // If it's a function or subscript with a parameter with leading
-  // double underscore, it's a private function or subscript.
-  if (isa<AbstractFunctionDecl>(D) || isa<SubscriptDecl>(D)) {
-    if (getParameterList(cast<ValueDecl>(D))->hasInternalParameter(Prefix))
-      return true;
-  }
-
-  if (auto *VD = dyn_cast<ValueDecl>(D)) {
-    auto Name = VD->getBaseName();
-    if (!Name.isSpecial() &&
-        Name.getIdentifier().str().startswith(Prefix)) {
-      return true;
-    }
-  }
-  return false;
-}
-
-static bool shouldIncludeDecl(Decl *D, bool ExcludeDoubleUnderscore) {
-  if (auto *VD = dyn_cast<ValueDecl>(D)) {
-    // Skip the decl if it's not visible to clients. The use of
-    // getEffectiveAccess is unusual here; we want to take the testability
-    // state into account and emit documentation if and only if they are
-    // visible to clients (which means public ordinarily, but
-    // public+internal when testing enabled).
-    if (VD->getEffectiveAccess() < swift::AccessLevel::Public)
-      return false;
-  }
-
-  // Skip SPI decls, unless we're generating a symbol graph with SPI information.
-  if (D->isSPI() && !D->getASTContext().SymbolGraphOpts.IncludeSPISymbols)
-    return false;
```

Comment:
@QuietMisdreavus In factoring out `getDocCommentSerializationTargetFor`, I removed the check for `SymbolGraphOpts.IncludeSPISymbols` here so that the check is consistent for both serialization and deserialization. I don't believe this should be an issue though, as we now are able to serialize SPI decl comments in the swiftsourceinfo. Any concerns with this change?

Reply:
Is there any way to remove SPI information from `.swiftdoc` files after this change? I'm wondering about potential leaks of implementation details if someone had a doc comment that wasn't meant for the light of day landing in a file like this.

Reply:
To be clear, with this change there *isn't* any SPI information in swiftdoc. Where as previously if `SymbolGraphOpts.IncludeSPISymbols` was `true` it could have been. Now it is *only* serialized in swiftsourceinfo.

But in any case, isn't that up to whether symbol graph outputs the symbol or not? If it does then the symbol is leaked, irrespective of the documentation being leaked.

Reply:
Ah, i misunderstood then. The symbol graph only has SPI information if you specifically ask for it, but i wasn't sure if it was going to inadvertently show up in other places where it wasn't desired. This makes sense, thanks!